RickMorty/app/src/main/java/com/tolganacar/rickmorty/adapter/RickMortyAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package com.tolganacar.rickmorty.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.navigation.Navigation
+import androidx.recyclerview.widget.RecyclerView
+import com.tolganacar.rickmorty.databinding.RecyclerRowBinding
+import com.tolganacar.rickmorty.model.Character
+import com.tolganacar.rickmorty.util.downloadFromUrl
+import com.tolganacar.rickmorty.util.placeholderProgressBar
+import com.tolganacar.rickmorty.view.FeedFragmentDirections
+
+class RickMortyAdapter(val characterList: ArrayList<Character>): RecyclerView.Adapter<RickMortyAdapter.RickMortyHolder>() {
+
+    class RickMortyHolder(val binding : RecyclerRowBinding): RecyclerView.ViewHolder(binding.root){
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RickMortyHolder {
+        val binding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.context),parent,false)
+        return RickMortyHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: RickMortyHolder, position: Int) {
+        holder.binding.recyclerName.text = characterList.get(position).name
+        holder.binding.recyclerStatus.text = characterList.get(position).status
+        holder.binding.cardView.setOnClickListener {
+            val action = FeedFragmentDirections.actionFeedFragmentToDetailsFragment()
```

Comment:
`holder.binding.cardView.setOnClickListener` buradaki gibi listenerların aksiyonlarını adapter içerisine koymayız. Adapter yalnızca modeli xml'e bind eder ve kullanıcının aldığı aksiyonları view'a bildirir. Burada onclick listener setleyebilirsin ama click aksiyonunu adapter dinlemez.

Listenerlerın bulunması gereken yer view yani bu projede Fragment tarafı. Bunu da bir interface veya high order function ile yapabilirsin.

https://github.com/kkocaburak/crypto-currency-mvvm-android/blob/master/crypto-currency-app/app/src/main/java/com/bkarakoca/cryptocurrencyapp/scene/crypto/cryptolist/CryptoCoinListAdapter.kt

https://github.com/kkocaburak/crypto-currency-mvvm-android/blob/master/crypto-currency-app/app/src/main/java/com/bkarakoca/cryptocurrencyapp/scene/crypto/cryptolist/CryptoCoinListFragment.kt

Örnek için yukarıdaki linklere bakabilirsin.